src/main/java/command/AddDeadlineCommand.java

```diff
@@ -0,0 +1,75 @@
+package command;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+
+import andelu.AndeluException;
+import andelu.DateTimeManager;
+import andelu.Storage;
+import andelu.TaskList;
+import andelu.Ui;
+import task.Deadline;
+import task.Task;
+
+
+
+/**
+ * A AddDeadlineCommand class to add new Deadline task.
+ * A subclass of Command class.
+ */
+public class AddDeadlineCommand extends Command {
+
+    /** The information from the user to create new Deadline. */
+    private String input;
+
+    /**
+     * Creates a constructor with userInput as argument.
+     *
+     * @param userInput The information when creating new Deadline.
+     */
+    public AddDeadlineCommand(String userInput) {
+        input = userInput;
+    }
+
+    /**
+     * Adds new Deadline task.
+     * Performs some prior checks to ensure the validity of the new Deadline.
+     * If invalid input occurs, error message is returned.
+     * Returns the response from Andelu.
+     * @param tasks The TaskList Object that contains a List of Task.
+     * @param ui The Ui Object that interact with the user.
+     * @param storage Storage Manager to writing to the file.
+     * @return The response from Andelu.
+     * @throws AndeluException If there is missing description or invalid date and time for 'by'.
+     */
+    @Override
+    public String executeCommand(TaskList tasks, Ui ui, Storage storage) throws AndeluException {
+        String[] splitInput = input.split(" ");
+        if (splitInput.length <= 1) {
+            throw new AndeluException("Missing the description!");
+        }
+        String[] deadlineSplit = input.split("/");
+        if (deadlineSplit.length < 2) {
+            throw new AndeluException("Invalid format for new Deadline!");
+        }
+        String name = deadlineSplit[0].substring(9).trim();
+        String by = deadlineSplit[1].substring(3).trim();
```

Comment:
### General
Interesting way of parsing! I may consider using this in my project as well. Thanks for the inspiration!